docs/guides/1238-project-file-organization.md

```diff
@@ -0,0 +1,242 @@
+---
+slug: /1238/project-file-organization
+displayed_sidebar: "0.2"
+---
+
+# Project file organization
+
+As time will pass, your actions will get bigger and bigger. You will feel the need to better organize your file structure by moving some parts in subfolders.
+
+The easiest way, when iterating fast, is to rely on the root module.
+
+## CUE Module
+
+A module in CUE is any directory including a `cue.mod` folder. It makes for the prefix/root of importable packages (e.g, `universe.dagger.io` is a module)
+
+### Anonymous module (default)
+
+The default module name is an empty string, which means it's anonymous. That's when you don't need to import other packages inside your module. It represents most cases where someone has a Dagger plan and imports everything from third-party packages, like universe.
+
+### When to use a module
+
+You can add a different module name if you want to import other packages from inside your module. This is required so you have a prefix/root before your imports. This is when you want to split your code into multiple files.
+
+The module path becomes the import path prefix for packages in the module. This might be the name of a domain you own or another name you control (such as your company name), even your email, followed optionally by a descriptive path (e.g., project name).
+
+You import packages by prefixing the module name they're a part of, plus the path to them, relative to the `cue.mod` directory.
+
+```console
+root                    // <- this is a module because it includes a cue.mod dir
+|-- cue.mod
+|   |-- module.cue      // module: "example.com/pkg"
+|-- schemas
+|   |-- compose         // <- this is a package because it includes files with a package directive
+|   |   |-- spec.cue    // package compose
+...
+|-- plan.cue            // import "example.com/pkg/schemas/compose"
+```
+
+### Summary
+
+Consider the root module as the URL to access the root of your project. Any subfolder inside this module needs to have Cue files with a package name equivalent to the dirname. The filename inside each folder is not important, the package name is:
+
+```console
+$  ls
+bar       cue.mod   foo
+
+$ cat foo/main.cue
+cat foo/main.cue 
+package foo
+
+$ cat bar/anything.cue
+package bar
+```
+
+To reference any of the code lying inside each of these packages:
+
+```cue
+import(
+    "info@foo.bar/foo" // imports CUE code inside `/foo`
+    "info@foo.bar/bar" // imports CUE code inside `/bar`
+)
+```
+
+## Initializing the root module
+
+### Project non initialized
+
+Dagger holds a way to set a module name at project initialization: `dagger project init --name <NAME>`.
+
+As said above, the name has to at least follow the structure of a domain: `domain.extension` or `domain.extension/project`. Email addresses are also accepted, and the referenced domain doesn't necessarly have to exist.
```

Comment:
`As mentioned above, the module name...`

`Email addresses can also be used for the module name, and the referenced...`